src/Database/Schema/Collection.php

```diff
@@ -53,14 +58,51 @@ public function __construct(Connection $connection)
         $this->_dialect = $connection->getDriver()->schemaDialect();
     }
 
+    /**
+     * Get the list of tables, excluding any views, available in the current connection.
+     *
+     * @return array<string> The list of tables in the connected database/schema.
+     */
+    public function listTablesWithoutViews(): array
+    {
+        [$sql, $params] = $this->_dialect->listTablesWithoutViewsSql($this->_connection->config());
+        $result = [];
+        $statement = $this->_connection->execute($sql, $params);
+        while ($row = $statement->fetch()) {
+            $result[] = $row[0];
+        }
+        $statement->closeCursor();
+
+        return $result;
+    }
+
     /**
      * Get the list of tables available in the current connection.
      *
+     * @deprecated in 4.3.3 Use {@link listTablesAndViews()} instead.
      * @return array<string> The list of tables in the connected database/schema.
      */
     public function listTables(): array
     {
-        [$sql, $params] = $this->_dialect->listTablesSql($this->_connection->config());
+        [$sql, $params] = $this->_dialect->listTablesAndViewsSql($this->_connection->config());
+        $result = [];
+        $statement = $this->_connection->execute($sql, $params);
+        while ($row = $statement->fetch()) {
+            $result[] = $row[0];
+        }
+        $statement->closeCursor();
+
+        return $result;
+    }
+
+    /**
+     * Get the list of tables available in the current connection.
+     *
+     * @return array<string> The list of tables in the connected database/schema.
+     */
+    public function listTablesAndViews(): array
```

Comment:
I'm wondering if we need to add this method as `listTables()` already exists and does the same thing. We could resolve the root problem by updating `ConnectionHelper` to use the new view-less methods.